client/plots/test/oncomatrix.spec.js

```diff
@@ -348,7 +376,75 @@ tape('launch matrix using runproteinpaint with launchGdcMatrix', function (test)
 			holder.querySelectorAll('svg text').length > 200,
 			'should have the expected number of rendered svg text elements'
 		)
-		if (test._ok && matrix?.destroy) setTimeout(matrix.destroy, 1000)
+
+		const rowcount = topGeneNum + termlst.length
+		test.equal(
+			select(holder).selectAll('.sjpp-matrix-term-label-g').node().querySelectorAll('text').length,
+			rowcount,
+			`should render ${rowcount} <series> elements`
+		)
+
+		if (test._ok) setTimeout(matrix.destroy, 1000) // TODO why cannot directly run matrix.destroy()
+		test.end()
+	}
+})
+
+tape('top mutated genes from APOLLO-LUAD, CNV only', function (test) {
+	test.timeoutAfter(100000)
+	test.plan(2)
+	const holder = select('body').append('div').node()
+
+	const topGeneNum = 10
+
+	runproteinpaint({
+		holder,
+		noheader: 1,
+		debugmode: true,
+		launchGdcMatrix: true,
+		filter0: { op: 'in', content: { field: 'cases.project.project_id', value: ['APOLLO-LUAD'] } },
+		settings: {
+			matrix: {
+				maxGenes: topGeneNum,
+				maxSample: 10000
+			}
+		},
+		opts: {
+			matrix: {
+				debug: 1,
+				callbacks: {
+					'postRender.test': runTests
+				}
+			}
+		}
+	})
+
+	async function runTests(matrix) {
+		//remove the default CNV filter for GDC (GDC hides CNV by default)
+		await matrix.Inner.app.dispatch({
```

Comment:
rather than a dispatch after rendering, how to directly set cnv to be visible (and doing oncoprint) in runpp config?

Reply:
we do not have that option now, the CNV filter (defined in gdc.hg38.ts) is overridden in getPlotConfig()